pandas/tests/io/json/test_json_table_schema.py

```diff
@@ -237,6 +237,26 @@ def test_build_series(self):
 
         assert result == expected
 
+    def test_read_json_from_to_json_results(self):
+        # GH 32383
+        df = pd.DataFrame(
+            {
+                "_id": {"0": "0"},
+                "category": {"0": "Goods"},
+                "recommender_id": {"0": "3"},
+                "recommender_name_jp": {"0": "浦田"},
+                "recommender_name_en": {"0": "Urata"},
+                "name_jp": {"0": "博多人形（松尾吉将まつお よしまさ）"},
+                "name_en": {"0": "Hakata Dolls Matsuo"},
+            }
+        )
+        df_json = df.to_json()
+        df_dict = json.loads(df_json)
+        result = pd.read_json(df_json, dtype=False)
+        result.index = result.index.astype(str)
```

Comment:
Is this required as well?

Reply:
reindex is required, they yield diff index type

Reply:
shouldn't be changing the result. create an expected = from df with the different index type if necessary.

Reply:
changed